src/App.tsx

```diff
@@ -1,26 +1,145 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+
+import {
+  createTodo,
+  deleteTodoFromServer,
+  getTodos,
+  USER_ID,
+} from './api/todos';
+import { wait } from './utils/fetchClient';
+import { filteringTodos } from './utils/queueTodos';
+import { countItemsLeft } from './utils/countItemsLeft';
+import { ErrorMsg } from './utils/ErrorMsg';
+import { Todo } from './types/Todo';
+import { TodoFilter } from './utils/TodoFilter';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoFilter>(TodoFilter.ALL);
+  const [error, setError] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setAllTodos)
+      .catch(() => {
+        setError(ErrorMsg.LIST_LOAD_ERROR);
+      });
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const completedTodos: Todo[] = allTodos.filter(todo => todo.completed);
+  const queuedTodos: Todo[] = filteringTodos(allTodos, filter);
+  const itemsLeft: number = countItemsLeft(allTodos);
+
+  const setLoading = (todoId: number, load: boolean) => {
+    setAllTodos(() => {
```

Comment:
```suggestion
    setAllTodos((todos) => {
```
you can get todos as an argument in the function. It's a preferable way when you set new state based on current.

Also, you can map todos, for example:
`todos.map((todo) => todo.id === todoId ? {...todo, loading: load} : todo)`